### ISSUE TITLE
feat: support passing options to `TypedRouterOptions` when using `experimental.typedPages`

### ISSUE THREAD

**Issue by danielwaltz** [nuxt/nuxt#27414]
### Describe the feature

When using the `experimental.typedPages` option with `unplugin-vue-router`, I ran into a case where the dynamically generated route names collided, leading to some route type names not being generated.

To get around this issue, it appears that `unplugin-vue-router` exposes an option called `getRouteName` in [its configuration](https://uvr.esm.is/guide/configuration.html) to customize the generated naming convention.

To take advantage of this, I figured it would be nice if the `experimental.typedPages` option also accepted an object of options in addition to a boolean that would be passed to `createRoutesContext`. Imagining something like this:

```ts
// nuxt.config.ts
export default defineNuxtConfig({
  experimental: {
    typedPages: {
  	  getRouteName: (routeNode) => customNameLogic(routeNode),
	},
  },
});
```

Would be happy to investigate and even implement a solution here, but wanted to share and gather opinions on if this is something the Nuxt team would be on board with. Thanks!

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by manniL** [comment]
Can you provide a reproduction where/when route names collide? Does that also happen *without* `typedPages`?

Unfortunately, functions are not allowed at all in the `nuxt.config.ts` to make sure the config is serializable but probably another way (like `router.options.ts`) would be an option for configuration

**Comment by danielwaltz** [comment]
@manniL Absolutely! Here's a minimal as I could make it reproduction of the issue.

https://stackblitz.com/edit/nuxt-starter-1zwqrx?file=nuxt.config.ts

For context, I created logical route groups for `/account/admin` and `/account/user`, where the goal is to not have the paths `/admin` and `/user` appear in the url, instead having their child pages show directly under `/account`. You can see this working without any changes in the example.

To reproduce the issue with `typedPages`, enable the option and open the `utils/directory.ts` file and swap the commented blocks of code. You should see type errors in the directory map, and if you inspect the keys on the `RouteNamedMap` type from `vue-router/auto/routes` you'll see that some pages are missing (presumably because they are using the same path).

This is why I think it would be useful to provide custom logic to generate the name, so I can use more info than just the path to define what is a unique route.

I get this is kind of a unique case born from a desire for logical grouping of routes, but I do still think it could be useful to provide custom config for other use cases (and this could help support creating a nuxt module that enables support for logical page groups which I have WIP code for).

**Comment by danielroe** [comment]
Can you use `definePageMeta({ name: 'something' })` instead?

**Comment by danielwaltz** [comment]
That's good insight, thanks for the clarification on concerns. That does make me think this is more on the side of Nuxt then!

We may want to close this issue in favor of something that enables people to hook into the route generation to control how things like the route name are generated, if that is a desirable or acceptable alternative.

Unfortunately I did try manually setting names using `definePageMeta({ name: 'something' })` but the key of `NamedRouteMap` type from `vue-router/auto/routes` does not appear to align with those.

I forked my original reproduction to demonstrate this. You'll see that even though `index.vue` manually has a name of `home` defined, the key on `NamedRouteMap` is still `index`.

https://stackblitz.com/edit/nuxt-starter-t6bgym?file=utils%2Fdirectory.ts

**Comment by danielroe** [comment]
There is something that lets you hook into route names are generated - you can use `pages:extend`. Here's a example in the docs: https://nuxt.com/docs/guide/recipes/custom-routing#pages-hook.

But it sounds like there might be something else - I'll check your repro.

**Comment by danielwaltz** [comment]
Thanks for taking time to investigate, inlining the route names instead of using an auto import did help the name key generation aspect! 🎉 

However, the original issue of there likely being a path collision is still present. Here's yet another fork to demonstrate the issue: https://stackblitz.com/edit/nuxt-starter-wkcepg?file=utils%2Ftest.ts

I set up a module to log each route name generated by Nuxt. If you compare that list with the list in `NamedRouteMap` you should see a route missing, and the manual paths are not respected. I would guess this is related to trying to have two pages share the same path.

I feel a bit like I'm taking you on a wild goose chase, and I really don't feel this is an urgent issue compared with all the other exciting stuff happening right now. So I especially appreciate the time here and bearing with me!

**Comment by CodeMan62** [comment]
will try to investigate it and create a PR

**Comment by danielroe** [comment]
If there's a path collision, this is probably something we need to fix in Nuxt rather than passing options to TypedRouterOptions.

We don't want the names (that nuxt knows about) to diverge from the names that `unplugin-vue-router` knows about.

**Comment by danielroe** [comment]
closing as this is a bug now resolved by https://github.com/nuxt/nuxt/pull/35788.

thank you for flagging it up ❤️
